sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/Column.java

```diff
@@ -113,6 +125,12 @@ static Column create(
   @Nullable
   String generationExpression();
 
+  /**
+   * Returns the identity column specification of this table column. Null means no identity column.
+   */
+  @Nullable
+  IdentityColumnSpec identityColumnSpec();
```

Comment:
to double check, only one of the `defaultValue`, `generationExpression` and `identityColumnSpec` can be non-null, right?

Reply:
Great catch! We blocked generation expression to be specified with identity column in SQL syntax but we did not block `defaultValue` to be specified with identity column. Will provide a fix in this PR.